pr2eus/robot-interface.l

```diff
@@ -44,6 +44,58 @@
   (:time-to-finish ()
     (ros::ros-debug "[~A] time-to-finish ~A" ros::name-space time-to-finish)
     time-to-finish)
+  (:action-result-cb (msg)
+    "
+    display error messages from [depending on robots]/follow_joint_trajectory/result
+    "
+    ;; The following error messages will not appear for the first (send *ri* :angle-vector)
+    ;; To display it, use :wait-interpolation or (ros::spin-once groupname)
+    (let ((status-num (send (send msg :status) :status))
+          (status-word '(PENDING
+                         ACTIVE
+                         PREEMPTED
+                         SUCCEEDED
+                         ABORTED
+                         REJECTED
+                         PREEMPTING
+                         RECALLING
+                         RECALLED
+                         LOST))
+          (status-text (send (send msg :status) :text))
+          (error-code nil)
+          (error-word '(SUCCESSFUL
+                        INVALID_GOAL
+                        INVALID_JOINTS
+                        OLD_HEADER_TIMESTAMP
+                        PATH_TOLERANCE_VIOLATED
+                        GOAL_TOLERANCE_VIOLATED)))
+      ;; status
+      (cond ((not (or (= status-num 3) (= status-num 8)))
+             (ros::ros-error "joint trajectory status: ~A~%"
+                             (cons status-num (elt status-word status-num)))
+             (unless (string= "" status-text)
+               (ros::ros-error "~A~%" status-text)))
+            ((= status-num 8)
+             (ros::ros-warn "joint trajectory status: ~A~%"
+                            (cons status-num (elt status-word status-num)))
+             (unless (string= "" status-text)
+               (ros::ros-warn "~A~%" status-text))))
+      ;; error_code and error_string
+      (cond ((derivedp msg control_msgs::FollowJointTrajectoryActionResult)
+             ;; e.g. pr2_controllers_msgs/JointTrajectoryActionResult does not have :error_code
+             (setq error-code (send (send msg :result) :error_code))
+             (when (< error-code 0)
+               (ros::ros-error "error_code: ~A~%"
+                               (cons error-code (elt error-word (* -1 error-code))))
+               (unless (string= "" (send (send msg :result) :error_string))
+                 (ros::ros-error "~A~%" (send (send msg :result) :error_string)))))
+            ((= status-num 3)) ;; when succeeded not displaying warning
```

Comment:
In my opinion, if status is succeeded but error-string is not empty, we should display the error-string.

In addition, if you want to check `status-num`,  the check should be done in the first cond clause. (under the `;; status` line)

Reply:
Thank you for advice.
> In my opinion, if status is succeeded but error-string is not empty, we should display the error-string.

I think so, too.
I was concerned about displaying a warning about the result message type even when command succeeded.

Reply:
OK

I said

> In my opinion, if status is succeeded but error-string is not empty, we should display the error-string.

because the field name is `error_string`.

I think programmer should be able to see all errors.

Reply:
Sorry, I misunderstand what you said.
Please forget this line.
> In my opinion, if status is succeeded but error-string is not empty, we should display the error-string.

You can write like below to check `status-num` in the first cond clause
```
(cond ((= status-num 3)
  (send-super :action-result-cb msg)
  (return-from :action-result-cb))
```

Reply:
Thank you for advice.
I solved this problem by this code.